tumblr/kanvas-ios
# Adds support for swift package manager

## What it does

- Adds a `Package.swift` file.
- Moves resources to the `Resources/ `folder to be compatible with spm.

## TODO:
- Create a test target that can be run with spm. 
- Add a fastlane step to run the tests using spm.
- This is being worked on in this branch -> [test-target-spm](https://github.com/tumblr/kanvas-ios/tree/adriana/test-target-spm). It is not a straight forward migration because all the places where we access the Bundle need to be updated, and the snapshot library behaves differently when running the snapshots from SPM. They get stored in derived data instead of the correct folder. 

## How to test

- Open Example/ and run pod install
- Open KanvasExample.xcworkspace and run all tests using iPhone SE. They should pass.
- Open the `Package.swift`. You should be able to build it successfully.
